### ISSUE TITLE
Details about training dataset for synthesizer and vocoder

### ISSUE THREAD

**Issue by xw1324832579** [CorentinJ/Real-Time-Voice-Cloning#61]
@CorentinJ Hello, I am confused with the component of training dataset for synthesizer and vocoder. Is it trained on the whole librispeech (clean and other, totally 960) or only on librispeech train-other-500? I haven't find the details about this in your thesis, maybe I missed something please tell me.
What's more, the effect of the system contains naturalness and similarity, is this affected by the component of training dataset? For example, if A is similar to B, whether A appeared in training dataset will affect the synthesizing effect of B, and if no one similar to B has appeared in training dataset, the synthesizing effect of B will be terrible. Is that right?

**Comment by CorentinJ** [comment]
The synthesizer/vocoder are trained on the clean sets (360 + 100). Including the other-500 set could be beneficial to generalization, not sure.

And yes the framework is very much limited by the ability of the synthesizer to generalize, which it's no doing so well. Voices with a non-English accent are very poorly cloned for example.

**Comment by xw1324832579** [comment]
Thank you for replying me in detail. And I have another question ,how do you get the EER 4.5% in your tests in speaker encoder, which test dataset(which speakers and which utterances) do you use? I haven't find this in your thesis, and I guess you use the same dataset as the author, right? However, after reading the paper, I find the author use 21 speakers from VCTK and Librispeech-clean using 100 test utterances each speaker, but  totally get 21,000 or 23,100 trials for test, that means 1000 utterances for each speaker, how many utterances do you use, and these get from raw wav in original dataset or synthesized wavs getting from the system?

**Comment by fazlekarim** [comment]
Check his thesis. I believe he wrote which datasets he used.

**Comment by CorentinJ** [comment]
[Starting from the last paragraph of page 19](https://matheo.uliege.be/bitstream/2268.2/6801/5/s123578Jemine2019.pdf#page=20)

**Comment by xw1324832579** [comment]
@CorentinJ Sorry, I missed details in this part in your thesis before. However, after doing tests following this part, using your pretrained model I still can't reproduce your result that EER is 4.53%.Please help me check my process for testing. 
First, I preprocess the libirspeech-test-clean、librispeech-test-other、VoxCeleb1、2 test datasets and get 231 speakers totally, then following the demo_cli.py to get embeddings for each speaker using 6 utterances each, and then joint them to get an embedding array shapes (231,6,256). At last, I use the function "loss" in encoder/model.py to compute EER, and I get 1.37% which has a big difference between you. Which of my steps is wrong?

**Comment by CorentinJ** [comment]
Yeah so as I stated in my thesis I have no good grounds for comparison and whatever the number I reported are, they don't mean much because the authors of SV2TTS give next to no info on how they evaluate the EER, which is a delicate procedure. I can give you my script, but I don't know if will help you find the answer you're looking for.

```
from encoder.data_objects import SpeakerVerificationDataLoader, SpeakerVerificationDataset
from encoder.model import SpeakerEncoder
from pathlib import Path
import matplotlib.pyplot as plt
import numpy as np
import torch


# This is my script for computing the test EER.
dataset_root = r"E:\Datasets\SV2TTS\encoder_test"

if __name__ == '__main__':
    speakers_per_batch = 32
    steps = 100
    
    dataset = SpeakerVerificationDataset(Path(dataset_root))
    
    model = SpeakerEncoder(torch.device("cuda"), torch.device("cpu"))
    checkpoint = torch.load("saved_models/pretrained.pt")
    model.load_state_dict(checkpoint["model_state"])
    model.eval()
    
    results = []
    for utterances_per_speaker in range(6, 8):
        loader = SpeakerVerificationDataLoader(
            dataset,
            speakers_per_batch=speakers_per_batch,
            utterances_per_speaker=utterances_per_speaker,
            num_workers=8,
        )
        with torch.no_grad():
            eers = []
            for step, speaker_batch in zip(range(1, steps + 1), loader):
                inputs = torch.from_numpy(speaker_batch.data).cuda()
                embeds = model(inputs)
                embeds_loss = embeds.view((speakers_per_batch, utterances_per_speaker, -1)).cpu()
                _, eer = model.loss(embeds_loss)
                
                eers.append(eer)
                print("Step %d    EER: %.3f" % (step, np.mean(eers)))
        results.append(np.mean(eers))
        
    plt.plot(range(2, 11), results)
    plt.xlabel("Enrollment utterances")
    plt.ylabel("Equal Error Rate")
    plt.show()
```
